File: packages/logger-android/src/main/kotlin/com/codex/logger/LogLevel.kt
package com.codex.logger

enum class LogLevel {
    DEBUG,
    INFO,
    WARN,
    ERROR,
    FATAL;

    fun isEnabled(minimum: LogLevel): Boolean = ordinal >= minimum.ordinal

    fun wireName(): String = when (this) {
        DEBUG -> "D"
        INFO -> "I"
        WARN -> "W"
        ERROR -> "E"
        FATAL -> "F"
    }
}

File: packages/logger-ios/Sources/LogLevel.swift
public enum LogLevel: Int, Sendable {
    case debug
    case info
    case warn
    case error
    case fatal

    func isEnabled(minimum: LogLevel) -> Bool {
        rawValue >= minimum.rawValue
    }

    func wireName() -> String {
        switch self {
        case .debug: return "D"
        case .info: return "I"
        case .warn: return "W"
        case .error: return "E"
        case .fatal: return "F"
        }
    }
}
